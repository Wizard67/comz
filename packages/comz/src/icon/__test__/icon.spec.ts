import { mount } from '@vue/test-utils'
import Icon from '../icon.vue'

describe('Icon.vue', () => {
  describe('props', () => {
    it('size', () => {
      const wrapper = mount(Icon, {
        props: {
          size: '20px'
        }
      })

      expect(wrapper.attributes('style')).toBe('--cicon-size: 20px;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('color', () => {
      const wrapper = mount(Icon, {
        props: {
          color: 'red'
        }
      })

      expect(wrapper.attributes('style')).toBe('--cicon-color: red;')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('spin', () => {
      const wrapper = mount(Icon, {
        props: {
          spin: true
        }
      })

      expect(wrapper.classes('')).toContain('cicon--spin')
      expect(wrapper.element).toMatchSnapshot()
    })

    it('link', () => {
      const wrapper = mount(Icon, {
        props: {
          link: true
        }
      })

      expect(wrapper.classes('')).toContain('cicon--link')
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('event', () => {
    it('on-click', () => {
      const wrapper = mount(Icon)
      
      wrapper.trigger('click')
      expect(wrapper.emitted()).toHaveProperty('on-click')
    })
  })

  describe('slot', () => {
    const content = "icon"

    const wrapper = mount(Icon, {
      slots: {
        default: content
      }
    })

    expect(wrapper.text()).toContain(content)
    expect(wrapper.element).toMatchSnapshot()
  })
})
