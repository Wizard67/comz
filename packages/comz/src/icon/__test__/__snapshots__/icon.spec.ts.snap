// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="cicon"
>
  
  
  icon
  
  
</div>
`;

exports[`Icon.vue props color 1`] = `
<div
  class="cicon"
  style="--cicon-color: red;"
>
  
  
</div>
`;

exports[`Icon.vue props link 1`] = `
<div
  class="cicon cicon--link"
>
  
  
</div>
`;

exports[`Icon.vue props size 1`] = `
<div
  class="cicon"
  style="--cicon-size: 20px;"
>
  
  
</div>
`;

exports[`Icon.vue props spin 1`] = `
<div
  class="cicon cicon--spin"
>
  
  
</div>
`;

exports[`Icon.vue props strokeWidth 1`] = `
<div
  class="cicon"
  style="--cicon-stroke-width: 1;"
>
  
  
</div>
`;
